### PR TITLE
Update CUDA build configs

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 
1. Fix Nuget package build break caused by #6225
2. Delete Dockerfile.centos_gpu. It is not used anywhere.
3. Fix Linux CUDA 10.2 build error caused by glibc upgrade

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/win-gpu-cuda-11-pipeline.yml [hunk 1]
```diff
@@ -117,15 +118,12 @@ jobs:
...
     displayName: 'Run tests'
+ 
 
   - template: templates/component-governance-component-detection-steps.yml
     parameters :
-      condition : 'succeeded'
-
-  - task: mspremier.PostBuildCleanup.PostBuildCleanup-task.PostBuildCleanup@3
-    displayName: 'Clean Agent Directories'
-    condition: always()
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/win-gpu-cuda-11-pipeline.yml:131]: why was the cleanup step removed?

File: tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml [hunk 2]
```diff
@@ -649,7 +649,7 @@ jobs:
         sudo apt autoremove
         PATH=/usr/lib/jvm/jre-1.8.0-openjdk.x86_64/bin:${PATH}
         echo "Java Version"
-        java --version
+        java -version
```
Comment (edgchen1) [inline tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:652]: >java -version [](start = 8, length = 13)

did this fail before? should we have something like "set -e" for the script?

File: tools/ci_build/github/azure-pipelines/win-gpu-cuda-11-pipeline.yml (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/win-gpu-cuda-11-pipeline.yml:131]: We have enough large disk space to store all the things. We don't need to clean it.

File: tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml (hunk 2 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/templates/c-api-cpu.yml:652]: Yes, failed and ignored. 

We should add "set -e". I will address it in another PR if you don't mind.

File: tools/ci_build/github/azure-pipelines/win-gpu-cuda-11-pipeline.yml (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/win-gpu-cuda-11-pipeline.yml:131]: The disk has 340GB.

**Comment by jywu-msft** [comment]
I'm worried about removing cudnn_home option.
on some platforms (e.g. Nvidia JetPack), they are pre-installed locations and they are not the same.
the user would have to move cudnn libraries to cuda location (which may not be possible for all cases)

**Review by jywu-msft — DISMISSED** [review]
i'm worried about removing cudnn_home build option and how it could break some user's scenarios.

**Comment by snnn** [comment]
> I'm worried about removing cudnn_home option.
> on some platforms (e.g. Nvidia JetPack), they are pre-installed locations and cuda_home/cudnn_home are not the same.
> the user would have to move cudnn libraries to cuda location (which may not be possible for all cases)



> i'm worried about removing cudnn_home build option and how it could break some user's scenarios.

Reverted.

File: tools/ci_build/github/azure-pipelines/win-gpu-cuda-11-pipeline.yml (hunk 1 above)
Comment (snnn) [reply tools/ci_build/github/azure-pipelines/win-gpu-cuda-11-pipeline.yml]: Moved the changes to #6616

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**